### PR TITLE
closing & finish() unclear

### PR DESCRIPTION
**Pull request by shampson**
Fix for issue [#15](https://github.com/w3c/webrtc-quic/issues/15).

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -910,7 +910,21 @@ <h2>Methods</h2>
...
+                <li> If <var>stream</var>'s  <a>[[\QuicStreamState]]</a> is <code>closing</code> and
+                the stream has received a STREAM frame with the FIN bit set, run the following steps:
+                  <ol>
+                    <li>Set <var>stream</var>'s <a>[[\Readable]]</a> slot to
+                    <code>false</code>.</li>
+                    <li>Set <var>stream</var>'s <a>[[\QuicStreamState]]</a> slot to
+                    <code>closed</code>.</li>
+                  </ol>
+                </li>
+                <li>Else, run the following step:
```
Comment (shampson) [inline index.html]: Move this above to change state to closing (same below)

File: index.html [hunk 2]
```diff
@@ -895,8 +895,8 @@ <h2>Methods</h2>
             </dd>
             <dt><dfn><code>finish</code></dfn></dt>
             <dd>
-              <p>Intitiates the closing procedure for the
-              <code><a>RTCQuicStream</a></code>. It may be called
+              <p>Intitiates the closing procedure for the local
```
Comment (shampson) [inline index.html]: Take out "local" and "to stop writing"

File: index.html [hunk 3]
```diff
@@ -1195,7 +1230,7 @@ <h2><dfn>RTCQuicStreamState</dfn> Enum</h2>
                 Alternatively:</p>
                   <ol>
                      <li>A RST_STREAM frame has been sent and a RST_STREAM
-                     frame or a STREAM frame within the FIN flag set has been
+                     frame or a STREAM frame with the FIN flag set has been
                      received, OR</li>
                      <li>A STREAM frame with the FIN flag set has been sent
```
Comment (shampson) [inline index.html]: Update closed state - When receiving a RST_STREAM frame

File: index.html [hunk 4]
```diff
@@ -1171,16 +1199,23 @@ <h2><dfn>RTCQuicStreamState</dfn> Enum</h2>
...
+                     set has not yet been received and the <code>reset</code>
+                     method has been called. This causes a RST_STREAM frame to be
+                     queued for transmission, the read and write buffers to be
+                     cleared and the <a>[[\Readable]]</a> and <a>[[\Writeable]]</a>
+                     slots are set to <code>false</code>.</li>
+                     <li>A RST_STREAM frame or STREAM frame with the FIN flag
+                     set has not yet been received and the <code>finish</code>
+                     method has been called. This causes a STREAM frame with the
+                     FIN flag set to be queued for transmission.</li>
+                     <li> The QUIC stream has received a RST_STREAM frame. This causes
```
Comment (shampson) [inline index.html]: Remove the RST_STREAM frame being received.

**Comment by shampson** [comment]
@aboba I made the changes we discussed. Let me know if it looks good.

**Comment by shampson** [comment]
It looks like this is ready to merge, although I don't have write access. Could somebody else please merge this?

**Comment by robin-raymond** [comment]
We need to clarify that receiving the FIN bit isn't the point when readable becomes false. Receiving the FIN bit means fully reading out all the data available and the FIN bit has been received. This needs clarification.

**Pull request merged**